### ISSUE TITLE
Savefig shrinks image

### ISSUE THREAD

**Issue by skjerns** [matplotlib/matplotlib#8805]
### Bug report

**Bug summary**

When I save a figure my canvas is smaller as before. As I need to extract images from an interactive pyplot I cannot align them anymore afterwards.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
f = plt.Figure()
plt.plot([1,2,3])
f.savefig('test.png')
```

**Actual outcome**

Figure gets saved, figure shrinks. Subsequent figure is then a tiny bit smaller

**Expected outcome**

The figure gets saved, the canvas or figure size stays untouched.

**Matplotlib version**

  * Operating System: Win10 64
  * Matplotlib Version: 2.0.2
  * Python Version: 3.5.2
  * Spyder 3 using Qt5

**Comment by tacaswell** [comment]
Suspect this is a duplicate of https://github.com/matplotlib/matplotlib/issues/8736

**Comment by skjerns** [comment]
Indeed, the fix suggested in #8736 did fix the problem, but only if I do not set a costum DPI.

**Comment by tacaswell** [comment]
closing in favor of #8736
